### DISCUSSION TITLE
HTTP 模式用 Nginx 反向代理是不是无法获取到客户端的 IP 用作 EDNS

### DISCUSSION THREAD

**Discussion by CalmLong** [IrineSistiana/mosdns#53] [Q&A] [answered]
如题，用的是服务器的 IP，不能动态使用客户端的 IP

**Comment by IrineSistiana** [comment]
不能。

现在没有读  http 头的机制，比如 X-Real-IP 什么的。

以后加。

**Comment by CalmLong** [comment, marked as answer]
好想要。。。

**Reply by IrineSistiana** [reply]
这功能会有的。当时加 http 的时候忘记一旦反代 ecs 功能就报废了。

**Reply by IceCodeNew** [reply]
> 这功能会有的。当时加 http 的时候忘记一旦反代 ecs 功能就报废了。

可以考虑用 PROXY protocol 实现？这样也不需要特意去读 HTTP 头了，不仅可以解决 DoH 下的问题，DoT 也可以一并解决了

**Comment by CalmLong** [comment]
@IrineSistiana ~~好像不起作用了~~

1.16.1 版本的 `get_user_ip_from_header` 不起作用了，1.14.0 正常

**Reply by IrineSistiana** [reply]
试试最新版 

~老年人了,之前剥离插件功能的时候好多功能参数忘记了~ -_-|| 感谢反馈.

**Reply by CalmLong** [reply]
好了
